Source/Parallelization/WarpXComm.cpp

```diff
@@ -391,19 +419,19 @@ WarpX::UpdateAuxilaryDataSameType ()
     for (int lev = 1; lev <= finest_level; ++lev)
     {
         const amrex::Periodicity& crse_period = Geom(lev-1).periodicity();
-        const IntVect& ng = Bfield_cp[lev][0]->nGrowVect();
-        const DistributionMapping& dm = Bfield_cp[lev][0]->DistributionMap();
+        const IntVect& ng = m_fields.get("Bfield_cp", Direction{0}, lev)->nGrowVect();
+        const DistributionMapping& dm = m_fields.get("Bfield_cp", Direction{0}, lev)->DistributionMap();
 
         // B field
         {
             if (electromagnetic_solver_id != ElectromagneticSolverAlgo::None)
             {
-                MultiFab dBx(Bfield_cp[lev][0]->boxArray(), dm, Bfield_cp[lev][0]->nComp(), ng);
-                MultiFab dBy(Bfield_cp[lev][1]->boxArray(), dm, Bfield_cp[lev][1]->nComp(), ng);
-                MultiFab dBz(Bfield_cp[lev][2]->boxArray(), dm, Bfield_cp[lev][2]->nComp(), ng);
-                dBx.setVal(0.0);
-                dBy.setVal(0.0);
-                dBz.setVal(0.0);
+                MultiFab dBx(m_fields.get("Bfield_cp",Direction{0},lev)->boxArray(), dm,
+                             m_fields.get("Bfield_cp",Direction{0},lev)->nComp(), ng);
+                MultiFab dBy(m_fields.get("Bfield_cp",Direction{1},lev)->boxArray(), dm,
+                             m_fields.get("Bfield_cp",Direction{1},lev)->nComp(), ng);
+                MultiFab dBz(m_fields.get("Bfield_cp",Direction{2},lev)->boxArray(), dm,
+                             m_fields.get("Bfield_cp",Direction{2},lev)->nComp(), ng);
```

Comment:
`setVal` calls deleted by mistake, as noticed by @RemiLehe:
```suggestion
                MultiFab dBx(m_fields.get("Bfield_cp",Direction{0},lev)->boxArray(), dm,
                             m_fields.get("Bfield_cp",Direction{0},lev)->nComp(), ng);
                MultiFab dBy(m_fields.get("Bfield_cp",Direction{1},lev)->boxArray(), dm,
                             m_fields.get("Bfield_cp",Direction{1},lev)->nComp(), ng);
                MultiFab dBz(m_fields.get("Bfield_cp",Direction{2},lev)->boxArray(), dm,
                             m_fields.get("Bfield_cp",Direction{2},lev)->nComp(), ng);
                dBx.setVal(0.0);
                dBy.setVal(0.0);
                dBz.setVal(0.0);
```